openSUSE-2017.7.2/salt.spec

```diff
@@ -15,6 +15,25 @@
 # Please submit bugfixes or comments via http://bugs.opensuse.org/
 #
 
+%if 0%{?suse_version} >= 1320
+# SLE15
+%global build_py3   1
+%global build_py2   1
+%global default_py3 1
+%else
+%if 0%{?suse_version} == 1315
+# SLE12
+%global build_py3   1
+%global build_py2   1
+%global default_py3 0
+%else
+# RES7
+%global build_py3   0
+%global build_py2   1
+%global default_py3 0
+%endif
+%endif
+%define pythonX %{?default_py3: python3}%{!?default_py3: python2}
```

Comment:
I would rather call it `python_bin` and do the following:
```specfile
%define python_bin %{?default_py3: python3}%{!?default_py3: python}
```

Then (below else, where you use `if bla, else the-same-bla-but-no-3`), like:
```specfile
%if 0%{?default_py3}
python3 setup.py test --runtests-opts=-u
%else
python setup.py test --runtests-opts=-u
%endif
```

This instead:
```specfile
%{python_bin} setup.py test --runtests-opts=-u
```

...etc.

Reply:
Of course, all `pythonX` occurrences should go away, and they are actually fine to be replaced with the `%{py_major}` (from the comment above on other place).

Reply:
pythonX is not about the binary. It is used as a prefix to define a dependency. 

We have salt package which should require either python2-salt or python3-salt depending on the default python version on that distribution.

```
Requires: %{pythonX}-%{name} = %{version}-%{release}
```

But feel free to name it different.